components/datetime/src/mock/timezone.rs

```diff
@@ -0,0 +1,297 @@
+    pub metazone_id: Option<String>,
+    pub time_variant: Option<String>,
+    pub country_code: Option<String>,
+}
+
+impl MockTimeZone {
+    pub const fn new(
+        gmt_offset: GmtOffset,
+        time_zone_id: Option<String>,
+        metazone_id: Option<String>,
+        time_variant: Option<String>,
+        country_code: Option<String>,
+    ) -> Self {
+        Self {
+            gmt_offset,
+            time_zone_id,
+            metazone_id,
+            time_variant,
+            country_code,
+        }
+    }
+
+    pub fn try_new(
+        gmt_offset: GmtOffset,
+        time_zone_id: Option<String>,
+        metazone_id: Option<String>,
+        time_variant: Option<String>,
+        country_code: Option<String>,
+    ) -> Result<Self, DateTimeError> {
+        Ok(Self {
+            gmt_offset,
+            time_zone_id,
+            metazone_id,
+            time_variant,
+            country_code,
+        })
+    }
+}
+impl FromStr for MockTimeZone {
+    type Err = DateTimeError;
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let gmt_offset = GmtOffset::from_str(input)?;
+        Ok(Self {
+            gmt_offset,
+            time_zone_id: None,
+            metazone_id: None,
+            time_variant: None,
+            country_code: None,
+        })
+    }
+}
+
+impl TimeZoneInput for MockTimeZone {
+    fn gmt_offset(&self) -> GmtOffset {
+        self.gmt_offset
+    }
+
+    fn time_zone_id(&self) -> Option<&str> {
+        self.time_zone_id.as_ref().map(AsRef::as_ref)
+    }
+
+    fn metazone_id(&self) -> Option<&str> {
+        self.metazone_id.as_ref().map(AsRef::as_ref)
+    }
+
+    fn time_variant(&self) -> Option<&str> {
+        self.time_variant.as_ref().map(AsRef::as_ref)
+    }
+
+    fn country_code(&self) -> Option<&str> {
+        self.country_code.as_ref().map(AsRef::as_ref)
+    }
+}
+
+/// Determines which ISO-8601 format should be used to format a `GmtOffset`.
+pub(crate) enum IsoFormat {
+    /// ISO-8601 Basic Format.
+    /// Formats zero-offset numerically.
+    /// e.g. +0500, +0000
+    Basic,
+
+    /// ISO-8601 Extended Format.
+    /// Formats zero-offset numerically.
+    /// e.g. +05:00, +00:00
+    Extended,
+
+    /// ISO-8601 Basic Format.
+    /// Formats zero-offset with the ISO-8601 UTC indicator: "Z"
+    /// e.g. +0500, Z
+    UtcBasic,
+
+    /// ISO-8601 Extended Format.
+    /// Formats zero-offset with the ISO-8601 UTC indicator: "Z"
+    /// e.g. +05:00, Z
+    UtcExtended,
+}
+
+/// Whether the minutes field should be optional or required in ISO-8601 format.
+pub(crate) enum IsoMinutes {
+    /// Minutes are always displayed.
+    Required,
+
+    /// Minutes are displayed only if they are non-zero.
+    Optional,
+}
+
+/// Whether the seconds field should be optional or excluded in ISO-8601 format.
+pub(crate) enum IsoSeconds {
+    /// Seconds are displayed only if they are non-zero.
+    Optional,
+
+    /// Seconds are not displayed.
+    None,
```

Comment:
`None` is not a keyword. It's a variant value of a single enum called `Option`. Just like this enum has a variant called `None` :)

I don't think we need to avoid `None` in our enums.

Reply:
I agree that it's fine to use `None` as an enum variant as far as Rust goes, but I also like that `Never` is a term used in 402. I would like to try to be consistent with other tools in the same domain, so I'll probably switch it to `Never`.